configs/components/openssl-3.0.rb

```diff
@@ -196,6 +196,10 @@
   #   install_commands << "rm -f #{settings[:prefix]}/bin/{openssl,c_rehash}"
   # end
 
+  if settings[:runtime_project] =~ /agent/
+    install_commands << "rm -f #{settings[:prefix]}/bin/c_rehash"
+  end
+
```

Comment:
Since `agent-runtime-main` is the only consumer of openssl 3, maybe remove it unconditionally? That way when client tools, pe installer, etc migrate to openssl 3, we don't reintroduce the security issue?